src/overlays/actors/ovl_En_Dai/z_en_dai.c

```diff
@@ -22,58 +43,697 @@ const ActorInit En_Dai_InitVars = {
+            (D_80B3FCE4[i] >= this->skelAnime.animCurrentFrame)) {
+            return D_80B3FD14[i];
+        }
+    }
+    return this->unk_1D6;
+}
+
+s32 func_80B3E96C(EnDai* this, GlobalContext* globalCtx) {
+    Vec3f sp74;
+    Vec3f sp68;
+    Vec3f sp5C;
+    Vec3f sp50;
+
+    if ((DECR(this->unk_1DA) == 0) &&
+        ((this->unk_1DC == 0) || func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount))) {
+        switch (this->unk_1DC) {
+            case 0:
+                func_80B3E5DC(this, 2);
+                this->unk_1DC += 1;
+                break;
+
+            case 1:
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_SNOWSTORM_HARD);
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_WEATHER_TAG, this->actor.world.pos.x,
+                            this->actor.world.pos.y, this->actor.world.pos.z, 0x1388, 0x708, 0x3E8, 0);
+                func_80B3E5DC(this, 3);
+                this->unk_1DC += 1;
+                break;
+
+            case 2:
+                func_80B3E5DC(this, 4);
+                this->unk_1DC += 1;
+                break;
+
+            case 3:
+                func_80B3E5DC(this, 1);
+                this->unk_1F0 = D_80B3FBF0;
+                this->unk_1FC = D_80B3FBF0;
+                this->unk_1DC = 0;
+                this->unk_1DA = Rand_S16Offset(20, 20);
+                break;
+        }
+    } else if (this->unk_1DC == 2) {
+        sp50.x = sp50.y = 0.0f;
+        sp50.z = 200.0f;
+
+        sp68.x = sp68.z = 0.0f;
+        sp68.y = 6.0f;
+
+        sp5C.x = Math_SinS(this->unk_1D4) * 80.0f;
+        sp5C.z = Math_CosS(this->unk_1D4) * 80.0f;
+        sp5C.y = -40.0f;
+
+        Lib_Vec3f_TranslateAndRotateY(&this->unk_1E4, this->unk_1D4, &sp50, &sp74);
+        func_80B3DFF0(this->particles, sp74, sp68, sp5C, 0.03f, 0.04f, 0x10);
+    }
+
+    return 0;
+}
+
+void func_80B3EC10(f32 arg0, Vec3f arg1, f32 arg2, Vec3f arg3, f32 arg4, Vec3f* arg5) {
+    f32 temp_f0 = (arg4 - arg0) / (arg2 - arg0);
+
+    arg5->x = ((arg3.x - arg1.x) * temp_f0) + arg1.x;
+    arg5->y = ((arg3.y - arg1.y) * temp_f0) + arg1.y;
+    arg5->z = ((arg3.z - arg1.z) * temp_f0) + arg1.z;
+}
+
+s32 func_80B3EC84(EnDai* this) {
+    static f32 D_80B3FD2C[] = {
+        6.0f, 11.0f, 15.0f, 0.0f, 12.0f, 17.0f, 0.0f, 3.0f,
+    };
+
+    static f32 D_80B3FD4C[] = {
+        8.0f, 15.0f, 24.0f, 4.0f, 17.0f, 26.0f, 3.0f, 7.0f,
+    };
+
+    static s16 D_80B3FD6C[] = {
+        0x0001, 0x0001, 0x0001, 0x0002, 0x0002, 0x0002, 0x0003, 0x0003,
+    };
+
+    static Vec3f D_80B3FD7C[] = {
+        { 1.0f, 1.0f, 1.0f }, { 1.0f, 0.8, 0.8 },   { 1.0f, 1.1f, 1.1f }, { 1.0f, 1.3f, 1.3f }, { 1.0f, 0.7f, 0.9f },
+        { 1.0f, 0.8, 0.9f },  { 1.0f, 0.7f, 0.9f }, { 1.0f, 0.8, 0.9f },  { 1.0f, 1.0f, 1.0f },
+    };
+
+    s32 i;
+    s32 ret = false;
+
+    for (i = 0; i < ARRAY_COUNT(D_80B3FD6C); i++) {
+        if (this->unk_1DC == D_80B3FD6C[i]) {
```

Comment:
I think all this would look better as a chained conditional (provided it still matches), but up to you